code/modules/mob/living/living.dm

```diff
@@ -1167,3 +1167,35 @@
 	C.take_organ_damage(damage)
 	C.KnockDown(3 SECONDS)
 	C.visible_message("<span class='danger'>[C] crashes into [src], knocking them both over!</span>", "<span class='userdanger'>You violently crash into [src]!</span>")
+
+/mob/living/throw_impact(atom/hit_atom, throwingdatum)
+	. = ..()
+	if(!has_status_effect(STATUS_EFFECT_CHARGING) || has_status_effect(STATUS_EFFECT_IMPACT_IMMUNE))
+		return
+
+	var/hit_something = FALSE
+	if(ismovable(hit_atom))
+		var/atom/movable/AM = hit_atom
+		var/atom/throw_target = get_edge_target_turf(AM, dir)
+		if(!AM.anchored || ismecha(AM))
+			AM.throw_at(throw_target, 5, 12, src)
+			hit_something = TRUE
+	if(isobj(hit_atom))
+		var/obj/O = hit_atom
+		O.take_damage(150, BRUTE)
+		hit_something = TRUE
+	if(isliving(hit_atom))
+		var/mob/living/L = hit_atom
+		L.adjustBruteLoss(60)
+		L.KnockDown(12 SECONDS)
+		L.Confused(10 SECONDS)
+		shake_camera(L, 4, 3)
+		hit_something = TRUE
+	if(isturf(hit_atom))
+		var/turf/T = hit_atom
+		if(iswallturf(T))
+			T.dismantle_wall(TRUE)
+			hit_something = TRUE
+	if(hit_something)
+		visible_message("<span class='danger'>[src] slams into [hit_atom]!</span>", "<span class='userdanger'>You slam into [hit_atom]!</span>")
+		playsound(get_turf(src), 'sound/effects/meteorimpact.ogg', 100, TRUE)
```

Comment:
you may want to abstract this out into a prince_charge_impact proc. That way `throw_impact` proc for all living mobs isn't taken all up by a very specific use case.

Reply:
Yeah this also makes me realize that this will proc will most likely be run twice by anything higher scoped than mob/living, will adjust!